heartbeat/monitors/active/icmp/loop.go

```diff
@@ -92,13 +94,25 @@ func newICMPLoop() (*icmpLoop, error) {
 	// IPv4/IPv6 checking
 	conn4 := createListener("IPv4", "ip4:icmp")
 	conn6 := createListener("IPv6", "ip6:ipv6-icmp")
-
+	unprivilegedPossible := false
 	l := &icmpLoop{
 		conn4:    conn4,
 		conn6:    conn6,
 		recv:     make(chan packet, 16),
 		requests: map[requestID]*requestContext{},
 	}
+	if conn4 == nil && conn6 == nil {
+
+		switch runtime.GOOS {
+		case "darwin":
+		case "linux":
+			unprivilegedPossible = true
+			conn4 = createListener("Unprivileged IPv4", "udp4")
```

Comment:
Why not assign it directly to conn4 and conn6. I'm actually wondering why we don't assign it directly already on lin 99 / 100. I suggest to always use `l.conn4` after 98 to make sure we don't confuse the vars and update one but no the other etc.

Reply:
Agreed, changed for the whole function